features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/algorithm/keygen/SnowflakeKeyGenerateAlgorithm.java

```diff
@@ -113,7 +113,9 @@ private int getMaxVibrationOffset(final Properties props) {
     }
     
     private int getMaxTolerateTimeDifferenceMilliseconds(final Properties props) {
-        return Integer.parseInt(props.getOrDefault(MAX_TOLERATE_TIME_DIFFERENCE_MILLISECONDS_KEY, MAX_TOLERATE_TIME_DIFFERENCE_MILLISECONDS).toString());
+        int result = Integer.parseInt(props.getOrDefault(MAX_TOLERATE_TIME_DIFFERENCE_MILLISECONDS_KEY, MAX_TOLERATE_TIME_DIFFERENCE_MILLISECONDS).toString());
```

Comment:
Hi @shengdoupi 
`Integer.parseInt` may throw `NumberFormatException` (when value is not a number), so try catch is needed here and wrapped as `KeyGenerateAlgorithmInitializationException`

Reply:
> Hi @shengdoupi `Integer.parseInt` may throw `NumberFormatException` (when value is not a number), so try catch is needed here and wrapped as `KeyGenerateAlgorithmInitializationException`

@RaigorJiang Thanks for your CR! `Integer.parseInt` and `Long.parseLong` are used without try catch in many algorithms' initial check, such as ShardingAutoTableAlgorithm and JobRateLimitAlgorithm, so I'm not sure if it necessary to use try catch here. @Pace2Car Hi, thank you very much for your possible advice.

Reply:
OK, then we can submit a separate issue later to handle all `NumberFormatException`.
Are you interested in getting it done?

Reply:
> OK, then we can submit a separate issue later to handle all `NumberFormatException`.
> Are you interested in getting it done?

OK, I will submit a issue for this later.